rules/CodingStyle/Rector/Use_/RemoveUnusedAliasRector.php

```diff
@@ -183,28 +182,19 @@ private function shouldSkip(Use_ $use, Name $name, string $lastName, string $ali
             return true;
         }
 
-        return (bool) $this->betterNodeFinder->findFirstNext($use, function (Node $node) use (
-            $name,
-            $loweredAliasName
-        ): bool {
-            if ($node instanceof FullyQualified) {
-                $originalName = $node->getAttribute(AttributeKey::ORIGINAL_NAME);
-                if ($originalName instanceof Name) {
-                    $loweredOriginalName = strtolower($originalName->toString());
-                    $loweredOriginalNameNamespace = Strings::before($loweredOriginalName, '\\');
-                    return $loweredAliasName === $loweredOriginalNameNamespace;
-                }
-            }
-
-            if (! $node instanceof ClassConstFetch) {
+        return (bool) $this->betterNodeFinder->findFirstNext($use, function (Node $node) use ($loweredAliasName): bool {
+            if (! $node instanceof FullyQualified) {
                 return false;
             }
 
-            if (! $node->class instanceof Name) {
+            $originalName = $node->getAttribute(AttributeKey::ORIGINAL_NAME);
+            if (! $originalName instanceof Name) {
                 return false;
             }
 
-            return $node->class->toString() === $name->toString();
+            $loweredOriginalName = strtolower($originalName->toString());
+            $loweredOriginalNameNamespace = Strings::before($loweredOriginalName, '\\');
+            return $loweredAliasName === $loweredOriginalNameNamespace;
         });
```

Comment:
Could you extract this to a XFinder class? It seems it could be re-used elsewhere

Reply:
implemented https://github.com/rectorphp/rector-src/pull/785/commits/d0dc5061fc6bc95320d016ec45c1655e50ccf39d, add FullyQualifiedFromUseFinder service for it